packages/auto-approve/src/check-pr.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// eslint-disable-next-line node/no-extraneous-import
+import {ProbotOctokit} from 'probot';
+import {PullRequestEvent} from '@octokit/webhooks-definitions/schema';
+import {getChangedFiles} from './get-PR-info';
+import {logger} from 'gcf-utils';
+// type PullsListFilesResponseData = operations['pulls/list-files']['responses']['200']['application/json'];
+
+// This file manages the logic to check whether a given PR matches the config in the repository
+
+export interface ValidPr {
+  author: string;
+  title: string;
+  changedFiles?: string[];
+  maxFiles?: number;
+}
+
+//TODO: fix pr any type to correct type
+/**
+ * Checks that a given PR matches the rules in the auto-approve.yml file in the repository
+ *
+ * @param config the config in the repository
+ * @param pr the incoming PR
+ * @param octokit the Octokit instance on which to make calls to the Github API
+ * @returns true if PR matches config appropriately, false if not
+ */
+export async function checkPRAgainstConfig(
+  config: {rules: ValidPr[]},
+  pr: PullRequestEvent,
+  octokit: InstanceType<typeof ProbotOctokit>
+): Promise<Boolean> {
+  const repoOwner = pr.pull_request.head.repo.owner.login;
+  const prAuthor = pr.pull_request.user.login;
+  const repo = pr.pull_request.head.repo.name;
+  const prNumber = pr.number;
+  const title = pr.pull_request.title;
+
+  const validTypeOfPR = config.rules.find(x => x.author === prAuthor);
+
+  if (validTypeOfPR) {
```

Comment:
perhaps call this `rulesToValidateAgainst`? took me a second to realize that this represented rules for merging a PR.